DataFormats/Candidate/interface/LeafCandidate.h

```diff
@@ -33,6 +33,8 @@ namespace reco {
 
     typedef unsigned int index;
 
+    LeafCandidate() {}
```

Comment:
If I understand correctly, the variadic tmpl constructor serves as an "explicit"  default constructor as well before the ```LeafCandidate() {}```  is introduced in this PR. 
I suppose the newly introduced ```LeafCandidate() {}``` serves as a specialization to ```LeafCandidate(Args && ...args)```
The two appear the same now, so it should work ok (until there's a need to have the variadic implemented in a different way, but why would there be).

Did I get it right?
@Dr15Jones @wmtan